source/_cookbook/dim_and_brighten_lights.markdown

```diff
@@ -89,79 +85,63 @@ input_number:
-              script.turn_on
-            {% else %}
-              script.turn_off
-            {% endif %}{% endraw %}
-          data:
-            entity_id: script.light_bright_pause
-        
-    light_bright_pause:
-      sequence:
-        - delay:
-            milliseconds: 1
-        - service: script.turn_on
-          data:
-            entity_id: script.light_bright
-
-    light_dim:
-      sequence:
-        - service: light.turn_on
-          data_template:
-            entity_id: light.YOUR_LIGHT
-            brightness: >-
-              {% raw %}{% set current = state_attr('light.YOUR_LIGHT', 'brightness')|default(0)|int %}
-              {% set step = states('input_number.light_step')|int %}
-              {% set next = current - step %}
-              {% if next < states('input_number.light_minimum')|int %}
-                {% set next = states('input_number.light_minimum')|int %}
-              {% endif %}
-              {{ next }}{% endraw %}
-
-        - service_template: >
-            {% raw %}{% if state_attr('light.YOUR_LIGHT', 'brightness')|default(0)|int > states('input_number.light_minimum')|int %}
-              script.turn_on
-            {% else %}
-              script.turn_off
-            {% endif %}{% endraw %}
-          data:
-            entity_id: script.light_dim_pause
-        
-    light_dim_pause:
-      sequence:
-        - delay:
-            milliseconds: 1
-        - service: script.turn_on
-          data:
-            entity_id: script.light_dim
+  ramp_light:
+    alias: Ramp Light Brightness
+    description: Ramp light brightness up or down
+    fields:
+      direction:
+        description: "Direction to change brightness: up or down"
+        example: up
+      light:
+        description: Light entity_id
+        example: light.family_room_lamp
+    mode: restart
+    sequence:
+      - repeat:
+          while:
+            - condition: template
+              value_template: >
+                {% set br = state_attr(light, 'brightness')|int(0) %}
+                {% set mn = states('input_number.light_minimum')|int %}
+                {% set mx = states('input_number.light_maximum')|int %}
+                {{ direction == 'up' and br < mx or
+                   direction == 'down' and br > mn }}
+          sequence:
+            - service: light.turn_on
+              data_template:
+                entity_id: "{{ light }}"
+                brightness: >
+                  {% set br = state_attr(light, 'brightness')|int(0) %}
+                  {% set mn = states('input_number.light_minimum')|int %}
+                  {% set mx = states('input_number.light_maximum')|int %}
+                  {% set st = states('input_number.light_step')|int %}
+                  {% if direction == 'up' %}
+                    {{ [br + st, mx]|min }}
+                  {% else %}
+                    {{ [br - st, mn]|max }}
+                  {% endif %}
```

Comment:
Lets replace this with `brightness_step` or `brightness_step_pct`. There is no reason to calculate it anymore.

Reply:
Ah, good point. I've never used it, but now that you mention it, I do remember hearing about it.

Reply:
No, wait. This example uses `input_number`'s to limit the min & max values, so it does still have to calculate the new values anyway to be able to make sure the limits are not exceeded. If the example didn't do that then, yes, using `brightness_step` would be more optimal.

Reply:
Alright!